src/audio-worklet/free-queue/README.md

```diff
@@ -0,0 +1,33 @@
+# Free Queue 
+
+A lock-free ring buffer implementation for high-performance audio processing 
+designed to be used on top of Web Audio API.
+
+It is based upon single-producer and single-consumer concept, so it can assure 
+thread safe concurrency without locks and mutexes, when following conditions 
+are satisfied -
+1. There is only one producer, that is only one thread/worker is pushing data 
+into buffer.
+2. There is only one consumer, that is only one thread/worker is pulling data 
+out of buffer.
+
+## API
+
+```ts
+  // Constructor
+  FreeQueue(size: number, channelCount: number = 1)
+  // push data into FreeQueue
+  push(input: Float32Array[], blockLength: number): boolean
```

Comment:
I don't feel strongly about this, but it may be good to note what the return value means here.  Up to you.